climada/entity/exposures/base.py

```diff
@@ -462,8 +463,14 @@ def plot_raster(self, res=None, raster_res=None, save_tiff=None,
             ras_tiff.close()
         # make plot
         crs_epsg, _ = u_plot.get_transformation(self.crs)
-        xmin, ymin, xmax, ymax = self.gdf.longitude.min(), self.gdf.latitude.min(), \
-        self.gdf.longitude.max(), self.gdf.latitude.max()
+        if isinstance(crs_epsg, ccrs.PlateCarree):
+            xmin, ymin, xmax, ymax = u_coord.latlon_bounds(
+                self.gdf.latitude.values, self.gdf.longitude.values)
+            mid_lon = 0.5 * (xmin + xmax)
+            crs_epsg = ccrs.PlateCarree(central_longitude=mid_lon)
+        else:
+            xmin, ymin, xmax, ymax = (self.gdf.longitude.min(), self.gdf.latitude.min(),
+                                      self.gdf.longitude.max(), self.gdf.latitude.max())
```

Comment:
Coming back to the `impact().plot_raster_eai_exposure()`, shouldn't fix to `plot_raster` should fix the problem of plotting impacts for exposures crossing the antimeridian?

Reply:
The code you're referring to here was a quick sketch. I don't have time today to test it. But maybe you are willing to do the testing? And maybe you can report back whether it also fixes `plot_raster_eai_exposure` for you?

Reply:
Unforuntaly the `plot_raster_eai_exposure` cannot deal with the change. For example I plotted the USA. Attached are the files before and after the change.

![Imp_USA_year2020_rcp26](https://user-images.githubusercontent.com/7040169/107958555-7a2bad80-6fa2-11eb-85c7-80bb25430731.png)

![Imp_USA_year2020_rcp26](https://user-images.githubusercontent.com/7040169/107958543-739d3600-6fa2-11eb-98ef-9c8545437cba.png)

Reply:
The output is not really helpful. The code to reproduce would be helpful.

Reply:
It is 
```
from climada.util.plot import make_map
fig, ax = make_map(num_sub=1, figsize=(9, 13))
imp.plot_raster_eai_exposure(axis=ax, label=xlabel)
```
where imp is an impact object with the USA as exposure.

To add to the above, not using a raster gives the correct map:

```
from climada.util.plot import make_map
fig, ax = make_map(num_sub=1, figsize=(9, 13))
imp.plot_hexbin_eai_exposure(axis=ax, label=xlabel)
```

![Imp_USA_year2020_rcp26s](https://user-images.githubusercontent.com/7040169/107959664-fbd00b00-6fa3-11eb-81da-ca099996c6e9.png)

Reply:
Would you please share a fully working example?

Reply:
```
import numpy as np
import pandas as pd
from scipy import sparse as sparse
from climada.entity import Exposures, IFTropCyclone, ImpactFuncSet
from climada.hazard import Hazard
from climada.engine import Impact

lon_list = np.append(np.arange(-170, -180.5, -1), np.arange(170, 180.5, 1))
lat_list = np.arange(-10, 10, 1)

df_exp = pd.DataFrame()
df_exp['latitude'] = np.repeat(lat_list, len(lon_list))
df_exp['longitude'] = np.repeat(lon_list, len(lat_list))
df_exp['value'] = 100
df_exp['if_TC'] = 1 
exp = Exposures(df_exp)
exp.set_geometry_points()
exp.check()

n_cen = len(exp.gdf.longitude) # number of centroids
n_ev = 10 # number of events
haz = Hazard('TC')
haz.centroids.set_lat_lon(exp.gdf.latitude, exp.gdf.longitude) # default crs used
haz.intensity = sparse.csr_matrix(np.random.random((n_ev, n_cen))*100)
haz.units = 'm'
haz.event_id = np.arange(n_ev, dtype=int)
haz.event_name = ['ev_12', 'ev_21', 'Maria', 'ev_35', 'Irma', 'ev_16', 'ev_15', 'Edgar', 'ev_1', 'ev_9']
haz.date = [721166, 734447, 734447, 734447, 721167, 721166, 721167, 721200, 721166, 721166]
haz.orig = np.zeros(n_ev, bool)
haz.frequency = np.ones(n_ev)/n_ev
haz.fraction = haz.intensity.copy()
haz.fraction.data.fill(1)
haz.check()
haz.centroids.plot()

exp.assign_centroids(haz)

impf = IFTropCyclone()
impf.set_emanuel_usa()
impf_set= ImpactFuncSet()
impf_set.append(impf)

imp = Impact()
imp.calc(exp, impf_set, haz)

imp.plot_hexbin_eai_exposure(pop_name=False)
imp.plot_raster_eai_exposure()

from climada.util.plot import make_map
fig, ax = make_map(num_sub=1, figsize=(9, 13))
imp.plot_raster_eai_exposure(axis=ax)
```

Reply:
Now the plot with `imp.plot_raster_eai_exposure()` is completely false since the raster is now made with renormalized longitudes. On the other hand `imp.plot_hexbin_eai_exposure()` gives a correct plot (since it does not generate a raster), but shows all longitudes (i.e. does not deal with the antimeridian).

Reply:
Thanks for the example, this is very helpful!

I fixed `Centroids.plot`, `Exposures.plot_raster` and `Exposures.plot_hexbin` for data crossing the antimeridian. However, the version in the last two lines will never work. In that case, the axis is provided by the user, so the user has already specified the projection of the axis (including the central longitude). The projection of an axis cannot be changed after initialization.

Reply:
Nice, that was quick! I guess we should now go through all the plotting functions. Or did you already check it?

Reply:
There are so many plotting functions in CLIMADA that don't deal with the antimeridian correctly. Going through all of them would become a really huge task. The main idea behind this PR was the `points_to_raster` functionality and for that, I already checked all affected plotting functions. I suggest to merge the PR and deal with the remaining plotting functions later.